### DISCUSSION TITLE
How would you download a file?

### DISCUSSION THREAD

**Discussion by suryavirkapur** [sagebind/isahc#372] [Questions] [answered]
Hi guys, I am new to Rust and the crate itself. I would really appreciate if you could give me a hint into how I could download a file using isahc.

**Comment by sagebind** [comment, marked as answer]
So downloading a file usually involves making a `GET` request, so you'd likely use [`get`](https://docs.rs/isahc/latest/isahc/fn.get.html) for that. Then you can pipe the response into a file using the [`copy_to_file`](https://docs.rs/isahc/latest/isahc/trait.ReadResponseExt.html#method.copy_to_file) extension method on the response. So an example might look like this:

```rust
use isahc::prelude::*;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
    // Open an HTTP connection to the server and start streaming the file.
    let mut response = isahc::get("https://example.org/some-file-to-download")?;

    // Check if the server returned a success code.
    if response.status().is_success() {
        // Pipe the response stream to a local file.
        response.copy_to_file("some-file-downloaded")?;
    } else {
        println!("Server returned status code {}", response.status());
    }

    Ok(())
}
```

**Comment by suryavirkapur** [comment]
Thank you.
